### PR TITLE
added event receiver for balance_responsible_parties_changed

### PR DESCRIPTION
**Pull request by xnetsrak**
## References

* #0000

### CODE REVIEW

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Common/Configuration/ConsumeServiceBusSettings.cs [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+namespace Energinet.DataHub.MarketParticipant.Common.Configuration;
+
+public class ConsumeServiceBusSettings
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Common/Configuration/ConsumeServiceBusSettings.cs:17]: Move this to Organization Host.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.EntryPoint.Organization/Integration/IntegrationEventSubscriptionHandler.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    private static Domain.Model.Events.BalanceResponsiblePartiesChanged Map(Guid eventId, BalanceResponsiblePartiesChanged balanceResponsiblePartiesChanged)
+    {
+        return new Domain.Model.Events.BalanceResponsiblePartiesChanged(
+            eventId,
+            ActorNumber.Create(balanceResponsiblePartiesChanged.EnergySupplierId),
+            ActorNumber.Create(balanceResponsiblePartiesChanged.BalanceResponsibleId),
+            new GridAreaCode(balanceResponsiblePartiesChanged.GridAreaCode),
+            balanceResponsiblePartiesChanged.Received.ToInstant(),
+            balanceResponsiblePartiesChanged.ValidFrom.ToInstant(),
+            balanceResponsiblePartiesChanged.ValidTo.ToInstant());
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.EntryPoint.Organization/Integration/IntegrationEventSubscriptionHandler.cs:50]: Does this extension method work on nullables, or is ValidTo never null?

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Events/BalanceResponsiblePartiesChanged.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using NodaTime;
+
+namespace Energinet.DataHub.MarketParticipant.Domain.Model.Events;
+
+public sealed class BalanceResponsiblePartiesChanged : DomainEvent
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Domain/Model/Events/BalanceResponsiblePartiesChanged.cs:20]: This is not a domain event. You can move this as a Dto to application and revert changes IntegrationEventProviderTests.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Integration/IBalanceResponsiblePartiesChangedEventHandler.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+using System.Threading.Tasks;
+using Energinet.DataHub.MarketParticipant.Domain.Model.Events;
+
+namespace Energinet.DataHub.MarketParticipant.Application.Handlers.Integration;
+
+#pragma warning disable CA1711
+public interface IBalanceResponsiblePartiesChangedEventHandler
+#pragma warning restore CA1711
+{
+    Task HandleAsync(BalanceResponsiblePartiesChanged balanceResponsiblePartiesChanged);
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.Application/Handlers/Integration/IBalanceResponsiblePartiesChangedEventHandler.cs:24]: The handler should not expect domain as input; either make Dto, or take Contract as input directly.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.IntegrationTests/Services/IntegrationEventProviderIntegrationTests.cs [hunk 5]
```diff
@@ -65,10 +66,13 @@ public async Task GetAsync_WithEvent_IntegrationEventReturned()
         await using var scope = host.BeginScope();
         await using var context = _fixture.DatabaseManager.CreateDbContext();
 
+        var ignoreDomainEventTypes = new List<string> { nameof(BalanceResponsiblePartiesChanged) };
+
         var allIntegrationEvents = typeof(DomainEvent)
             .Assembly
             .GetTypes()
             .Where(domainEventType => domainEventType.IsSubclassOf(typeof(DomainEvent)))
+            .Where(d => !ignoreDomainEventTypes.Contains(d.Name))
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.IntegrationTests/Services/IntegrationEventProviderIntegrationTests.cs:75]: You can revert this.

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.EntryPoint.Organization/Integration/IntegrationEventSubscriptionHandler.cs (hunk 2 above)
Comment (xnetsrak) [reply source/marketparticipant/Energinet.DataHub.MarketParticipant.EntryPoint.Organization/Integration/IntegrationEventSubscriptionHandler.cs:50]: "fixed", but I will make integration test for this event flow ..

File: source/marketparticipant/Energinet.DataHub.MarketParticipant.IntegrationTests/Services/IntegrationEventProviderIntegrationTests.cs [hunk 6]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 using System;
+using System.Collections.Generic;
```
Comment (defectiveAi) [inline source/marketparticipant/Energinet.DataHub.MarketParticipant.IntegrationTests/Services/IntegrationEventProviderIntegrationTests.cs:16]: Dead using

**Review by defectiveAi — APPROVED** [review]

**Pull request merged**